### PR TITLE
Fixes 4488: Add upload custom content modal

### PR DESCRIPTION
**Pull request by Andrewgdewar**
## Summary
This adds the following modal: 

Screenshot 2024-09-06 at 12 04 13 PM
Screenshot 2024-09-06 at 12 04 18 PM
Screenshot 2024-09-06 at 12 04 22 PM
Screenshot 2024-09-06 at 12 03 54 PM

## Testing steps
- Create a upload repository (split buttons added in this PR @swadeley)
- Navigate to the upload modal via the addModal button or the kebab in the contentList. 
- Upload a few rpms. 
- Confirm changes.

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMS-4488

**Comment by jlsherrill** [comment]
image

I think its a tad confusing to have a button that says "Upload now" when technically the files are already uploaded and you've shown them being upload.   Maybe  "Create Snapshot", "finalize uploads", or something like that would be a better wording?

**Comment by Andrewgdewar** [comment]
> image
> 
> I think its a tad confusing to have a button that says "Upload now" when technically the files are already uploaded and you've shown them being upload. Maybe "Create Snapshot", "finalize uploads", or something like that would be a better wording?

I agree!

**Comment by xbhouse** [comment]
when an upload repo is created, it looks like the alert pops up saying there is introspection data already available. since introspection should't occur for upload repos, maybe we should disable this alert or have it say something different? 
Screenshot 2024-09-10 at 3 12 28 PM

**Comment by xbhouse** [comment]
similar to Justin's comment, saying `x out of x files are ready to be uploaded` could be confusing since they've already been uploaded. maybe we could say `ready to be added to the repository` or `ready to be snapshotted` instead?

**Comment by xbhouse** [comment]
if i leave the upload modal up for a certain amount of time (at least 15 minutes), i see this error and need to refresh which loses all the uploaded content. maybe a patternfly or chrome issue?
```
Error: Failed to execute 'removeChild' on 'Node': The node to be removed is not a child of this node.
at Modal.componentWillUnmount (https://stage.foo.redhat.com:1337/apps/content-sources/js/vendors-node_modules_patternfly_react-core_dist_esm_components_Modal_index_js.js:148:20)
at dl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264932)
at hl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264023)
at dl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264220)
at hl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264023)
at dl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264782)
at hl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264023)
at dl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264220)
at hl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264023)
at dl (https://stage.foo.redhat.com:1337/apps/chrome/js/chrome-root.6c798ea788727869.js:2:264782)
```

File: src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx [hunk 1]
```diff
@@ -0,0 +1,331 @@
...
+
+    let uuid: string = '';
+    let created: string = '';
+    if (!error) {
+      try {
+        const res = await createUpload(file.size);
+        uuid = res.upload_uuid;
+        created = res.created;
+      } catch (err) {
+        error = 'Failed checksum validation: ' + (err as Error).message;
```
Comment (xbhouse) [inline src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx:214]: during a network outage i sometimes see this message: `Failed checksum validation: Network Error`. i don't think we need `Failed checksum validation` in the error message in this case

**Comment by xbhouse** [comment]
if the upload was not created due to an error or outage, the request to upload_chunk fails, note the missing upload uuid here: `https://stage.foo.redhat.com:1337/api/content-sources/v1.0/repositories/uploads//upload_chunk/` and a retry doesn't work. it looks like i need to remove the files and re-add them manually so an upload gets created first.

i *think* this could be fixed by calling both the create_upload and upload_chunk apis on a retry? what do you think?

File: src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx (hunk 1 above)
Comment (Andrewgdewar) [reply src/Pages/Repositories/ContentListTable/components/UploadContent/compontents/FileUploader.tsx:214]: Good catch, the retry logic needed some work!

**Comment by Andrewgdewar** [comment]
> similar to Justin's comment, saying `x out of x files are ready to be uploaded` could be confusing since they've already been uploaded. maybe we could say `ready to be added to the repository` or `ready to be snapshotted` instead?
Screenshot 2024-09-12 at 11 18 03 AM

How does this look?

**Comment by xbhouse** [comment]
> > similar to Justin's comment, saying `x out of x files are ready to be uploaded` could be confusing since they've already been uploaded. maybe we could say `ready to be added to the repository` or `ready to be snapshotted` instead?
> > Screenshot 2024-09-12 at 11 18 03 AM
> 
> How does this look?

looks good to me!

**Review by xbhouse — APPROVED** [review]
looks and works great!! 🚀

**Comment by mayurilahane** [comment]
Screenshot from 2024-09-13 23-36-14
Screenshot from 2024-09-13 23-36-45
Screenshot from 2024-09-13 23-37-46
Screenshot from 2024-09-13 23-37-28
Screenshot from 2024-09-13 23-37-26

**Comment by mayurilahane** [comment]
1. As @xbhouse said when we upload the repo it gives a pop up msg saying "custom repository "test_repo" added
image

2. After uploading the rpm's the button asks to "confirm changes?" 
image

**Comment by mayurilahane** [comment]
when i have an "invalid rpm"  shouldn't it is suppose to give me warning saying its an invalid rpm ?
or it can't process the rpm before uploading it ?

image

**Comment by Andrewgdewar** [comment]
> 1. As @xbhouse said when we upload the repo it gives a pop up msg saying "custom repository "test_repo" added
>    image
> 2. After uploading the rpm's the button asks to "confirm changes?"
>    image

This is as intended. 
Because of how the api works.

**Comment by Andrewgdewar** [comment]
> shouldn't it is suppose to give me warning saying its an invalid rpm ?
> or it can't process the rpm before uploading it ?

Currently it does not confirm that the repository is valid. 
It only checks the file type. 
@jlsherrill Any chance we would add that functionality?

**Comment by mayurilahane** [comment]
> > 1. As @xbhouse said when we upload the repo it gives a pop up msg saying "custom repository "test_repo" added
> >    image
> > 2. After uploading the rpm's the button asks to "confirm changes?"
> >    image
> 
> What is your question? This is as intended for now. The "Confirm changes" is because the uploaded items will not be added to your repo until you click that. As for the "Custom repository created" would you prefer it to specify something else? "Upload repository"? I did adjust the description based off of her feedback with @jlsherrill. I could update the description from "Custom" > "Upload"

Hi @Andrewgdewar the discussion about the "confirm changes" button was not concluded that's why i just wanted to mention that this is what i am seeing right now.
 other than that everything looks good to me.

_sorry for creating the confusion_

**Comment by jlsherrill** [comment]
we can't really validate that its a true rpm until you click that button, so what you see is expected and about the best we can do.

**Comment by mayurilahane** [comment]
LGTM!

**Comment by swadeley** [comment]
@Andrewgdewar pls rebase

**Comment by swadeley** [comment]
Hi @Andrewgdewar 

Re Upload button

I see `data-ouia-component-id="upload-button"` on the div for the grid

Can you move it to the button class under the div class="pf-v5-c-multiple-file-upload__upload" ?

i.e. where it currently says `data-ouia-component-id="OUIA-Generated-Button-secondary-1"`


Thank you

**Comment by swadeley** [comment]
Hi

I created an upload repo and added two RPMs. After the second one I noticed this pop up message say "uploaded to undefined":

`Success alert:One rpm successfully uploaded to undefined`

I wonder why it does not use the repo name.

**Comment by Andrewgdewar** [comment]
> Hi
> 
> I created an upload repo and added two RPMs. After the second one I noticed this pop up message say "uploaded to undefined":
> 
> `Success alert:One rpm successfully uploaded to undefined`
> 
> I wonder why it does not use the repo name.

This is based off of the value of the callback name.

data.repository_name < 

That should be defined in the response.

**Comment by Andrewgdewar** [comment]
> Hi @Andrewgdewar
> 
> Re Upload button
> 
> I see `data-ouia-component-id="upload-button"` on the div for the grid
> 
> Can you move it to the button class under the div class="pf-v5-c-multiple-file-upload__upload" ?
> 
> i.e. where it currently says `data-ouia-component-id="OUIA-Generated-Button-secondary-1"`
> 
> Thank you

Unfortunately it isn't possible.

**Comment by xbhouse** [comment]
> > Hi
> > I created an upload repo and added two RPMs. After the second one I noticed this pop up message say "uploaded to undefined":
> > `Success alert:One rpm successfully uploaded to undefined`
> > I wonder why it does not use the repo name.
> 
> This is based off of the value of the callback name.
> 
> data.repository_name <
> 
> That should be defined in the response.

looks like the task response was changed recently, and `repository_name` was changed to `object_name`. so the AddUploadResponse probably needs updating. this is a sample task response now:
```
{
            "uuid": "f5a39a1e-c4a5-4b45-b210-dc3b43d5ac27",
            "status": "completed",
            "created_at": "2024-10-07T20:51:45Z",
            "ended_at": "2024-10-07T20:51:45Z",
            "error": "",
            "org_id": "17684632",
            "type": "introspect",
            "object_type": "repository",
            "object_name": "test1",
            "object_uuid": "32c4cf3f-5add-40ba-8a9d-cf5079e48b07"
}
```

**Comment by swadeley** [comment]
Hi, looks good now:

```
One rpm successfully uploaded to upload-test
This repository will be snapshotted shortly
```

**Pull request merged**